extensions/amp-google-assistant-assistjs/0.1/assistjs-runtime-service.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The shared service used by all custom elements to execute runtime tasks like
+ * adding ports to the PortNetwork.
+ */
+
+import * as closure from '../../../third_party/closure-responding-channel/closure-bundle';
```

Comment:
Why do we need a separate service instead of combining with `FrameService`?

Reply:
They would be respondible for different tasks. FrameService should execute tasks that requires the interaction with underlying iframe(e.g. with Assistant Server), whereas RuntimeService should execute tasks about runtime like managing ports of different iframes or resizing container of iframes.